app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -48,7 +48,10 @@ class NavigationOverlayRobot {
     // against any of its views
     fun assertOverlayIsOpen() = urlBar().assertIsDisplayed()
 
-    fun assertCanGoBack(canGoBack: Boolean) = backButton().assertIsEnabled(canGoBack)
+    fun assertCanGoBack(canGoBack: Boolean) {
+        device.wait(Until.findObject(By.res(R.id.tileContainer.toString())), 5000)
```

Comment:
CC @Baron-Severin 

I'm surprised that we have to add a manual wait. Is there any way to add inherent waits?

Reply:
I'm surprised too.  I believe there are already waits baked into Espresso for things like animations and such.  The back button activating is triggered by changes to the A-C `Session` object, so my best guess would be that those changes are not fast enough to be covered by the built-in system.  But this is really all conjecture, tbh.